lib/service/user.go

```diff
@@ -165,6 +165,19 @@ func (svc *LndhubService) CalcFeeLimit(destination string, amount int64) int64 {
 	return limit
 }
 
+func (svc *LndhubService) CheckUserBalance(ctx context.Context, amount, userId int64) (result *responses.ErrorResponse, err error) {
+	if svc.Config.MaxAccountBalance > 0 {
+		currentBalance, err := svc.CurrentUserBalance(ctx, userId)
+		if err != nil {
+			return nil, err
+		}
+		if currentBalance+amount > svc.Config.MaxAccountBalance {
+			return &responses.BadArgumentsError, fmt.Errorf("Max account balance exceeded for user_id:%d (balance:%d + amount:%d)", userId, currentBalance, amount)
```

Comment:
is this the best error to return here? 
or do we return a generic error that we then use in the controller?